sdcm/scan_operation_thread.py

```diff
@@ -307,28 +297,73 @@ def execute_query(self, session, cmd: str):
         session.default_consistency_level = ConsistencyLevel.ONE
         return session.execute_async(cmd)
 
-    def run_scan_operation(self, scan_operation_event, cmd: str = None):  # pylint: disable=too-many-locals
+    def reset_output_files(self):
+        self.normal_query_output.close()
+        self.reversed_query_output.close()
+        self.reversed_query_output = tempfile.NamedTemporaryFile(mode='w+', delete=False,  # pylint: disable=consider-using-with
+                                                                 encoding='utf-8')
+        self.normal_query_output = tempfile.NamedTemporaryFile(mode='w+', delete=False,  # pylint: disable=consider-using-with
+                                                               encoding='utf-8')
+
+    def _compare_output_files(self):
+        self.normal_query_output.flush()
+        self.reversed_query_output.flush()
+
+        with tempfile.NamedTemporaryFile(mode='w+', delete=True, encoding='utf-8') as reorder_normal_query_output:
+            cmd = f'tac {self.normal_query_output.name}'
+            if self.limit:
+                cmd += f' | head -n {self.limit}'
+            cmd += f' > {reorder_normal_query_output.name}'
+            LOCAL_CMD_RUNNER.run(cmd=cmd, ignore_status=True)
+            reorder_normal_query_output.flush()
+            LOCAL_CMD_RUNNER.run(cmd=f'cp {reorder_normal_query_output.name} {self.normal_query_output.name}',
+                                 ignore_status=True)
+            self.normal_query_output.flush()
+
+        file_names = f' {self.normal_query_output.name} {self.reversed_query_output.name}'
+        diff_cmd = f"diff -y --suppress-common-lines {file_names}"
+        self.log.info("Comparing scan queries output files by: %s", diff_cmd)
+        result = LOCAL_CMD_RUNNER.run(cmd=diff_cmd, ignore_status=True)
+        if not result.stderr:
+            if not result.stdout:
+                self.log.info("Compared output of normal and reversed queries is identical!")
+            else:
+                self.log.warning("Normal and reversed queries output differs: \n%s", result.stdout.strip())
+                ls_cmd = f"ls -alh {file_names}"
+                head_cmd = f"head {file_names}"
+                for cmd in [ls_cmd, head_cmd]:
+                    result = LOCAL_CMD_RUNNER.run(cmd=cmd, ignore_status=True)
+                    if result.stdout:
+                        stdout = result.stdout.strip()
+                    self.log.info("%s command output is: \n%s", cmd, stdout)
```

Comment:
stdout can be undefined, the log print should be inside the if

Reply:
fixed